internal/ctxval/logger.go

```diff
@@ -18,6 +20,8 @@ func Logger(ctx context.Context) *zerolog.Logger {
 }
 
 // WithLogger returns context copy with logger.
+//
+// Deprecated: Use log.Logger.WithContext(ctx) instead.
 func WithLogger(ctx context.Context, logger *zerolog.Logger) context.Context {
```

Comment:
We could easily drop this function and make sure we use `ctx := zerolog.WIthContext(ctx)` everywhere instead.